hydromt_wflow/wflow.py

```diff
@@ -4567,13 +4567,21 @@ def write_grid(
         if not isdir(dirname(fn)):
             os.makedirs(dirname(fn))
         self.logger.info(f"Write grid to {fn}")
-        mask = ds_out[self._MAPS["basins"]] > 0
-        for v in ds_out.data_vars:
-            # nodata is required for all but boolean fields
-            if ds_out[v].dtype != "bool":
-                ds_out[v] = ds_out[v].where(mask, ds_out[v].raster.nodata)
+        ds_out = self._mask_grid_to_basins(ds_out)
```

Comment:
The delineation of the basin is actually the very first thing that should be done based on the region. unless we change this behavior in wflow v1, masking on write should then not be needed anymore :)